src/core/public/chrome/ui/header/collapsible_nav.tsx

```diff
@@ -71,6 +71,48 @@ function getOrderedCategories(
   );
 }
 
+function getMergedNavLinks(
```

Comment:
This `getMergedNavLinks` function is a little bit hard to understand. Could we refactor it based below logic:
- separate `unknowns` to two groups, one for has order property, one for not
- combine `orderedCategories` and `categoryDictionary` to one array with category string and order, than separate two groups like `unknowns`
- combine `unknowns` and `new array` with order property in one array, than sort by order
- append all `unknowns` without order property
- append all `categories` without order property